core/src/main/java/com/datastax/oss/driver/internal/core/protocol/BuiltInCompressors.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.protocol;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.protocol.internal.Compressor;
+import io.netty.buffer.ByteBuf;
+
+public class BuiltInCompressors {
+
+  public static Compressor<ByteBuf> get(String name, DriverContext context) {
+    switch (name.toLowerCase()) {
+      case "lz4":
+        return new Lz4Compressor(context);
```

Comment:
Worth noting: this is immediately going to execute a (now completely redundant) runtime check of the classpath to validate that the exact same class we just checked for is present.  In a Graal context that check is _completely_ redundant.  Perhaps we should either expose some way to opt out of that check or maybe make the check contingent upon being in a non-Graal context within Lz4Compressor.java?

Reply:
Right. On the other hand it's just a `Class.forName` call, is it that significant at init time?

Reply:
Maybe the check done in `DependencyCheck` should be memoized.

Reply:
Isn't it executed only one time during the compilation phase?

Reply:
It'll get executed once at driver startup (possibly in native code depending on how the image was built).  To be clear it's a minor thing; it would be nice to not have to execute the same check twice but this is hardly a show-stopper.